File: Leet Code/First Bad Version/first_bad_version.cpp
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        long int first = 1;
        long int last = n;
        long int half;
        long int bad_version = n;
        
        while(first <= last){
            half =  (first + last) / 2;
            if(isBadVersion(half)){
                bad_version = half;
                last = half-1;
            }
            else{
                first = half+1;
            }
        }
        return bad_version;
        
        return 0;
    }
};

File: Leet Code/First Bad Version/first_bad_version.java
/* The isBadVersion API is defined in the parent class VersionControl.
      boolean isBadVersion(int version); */

public class Solution extends VersionControl {
    public int firstBadVersion(int n) {
        int low = 0, high = n;

        while(low<=high){
            int mid = (high+low)/2;

            if(isBadVersion(mid) && !(isBadVersion(mid-1)))
                return mid;
            else if(isBadVersion(mid)==false ) 
                low = mid+1;
            else 
                high= mid;
        }
        return -1;
    }
}
